crates/vizia_core/src/views/switch.rs

```diff
@@ -0,0 +1,205 @@
+/// ```
+///
+/// ## Switch with an action
+///
+/// A Switch can be used to trigger a callback when toggled. Usually this is emitting an
+/// event responsible for changing the data the Switch is bound to.
+///
+/// ```
+/// # use vizia_core::prelude::*;
+/// #
+/// # #[derive(Lens)]
+/// # struct AppData {
+/// #     value: bool,
+/// # }
+/// #
+/// # impl Model for AppData {}
+/// #
+/// # enum AppEvent {
+/// #     ToggleValue,
+/// # }
+/// #
+/// # let cx = &mut Context::new();
+/// #
+/// # AppData { value: false }.build(cx);
+/// #
+/// Switch::new(cx, AppData::value).on_toggle(|cx| cx.emit(AppEvent::ToggleValue));
+/// ```
+///
+/// ## Switch with a label
+///
+/// A Switch is usually used with a label next to it describing what data the Switch
+/// is bound to or what the Switch does when pressed. This can be done, for example, by
+/// wrapping the Switch in an [`HStack`](crate::prelude::HStack) and adding a [`Label`](crate::prelude::Label)
+/// to it.
+///
+/// ```
+/// # use vizia_core::prelude::*;
+/// #
+/// # #[derive(Lens)]
+/// # struct AppData {
+/// #     value: bool,
+/// # }
+/// #
+/// # impl Model for AppData {}
+/// #
+/// # let cx = &mut Context::new();
+/// #
+/// # AppData { value: false }.build(cx);
+/// #
+/// HStack::new(cx, |cx| {
+///     Switch::new(cx, AppData::value);
+///     Label::new(cx, "Press me");
+/// });
+/// ```
+///
+/// ## Custom Switch
+///
+/// The `text` modifier combined with a `map` on the lens can be used to customize the icon used by the Switch.
+///
+/// ```
+/// # use vizia_core::prelude::*;
+/// #
+/// # #[derive(Lens)]
+/// # struct AppData {
+/// #     value: bool,
+/// # }
+/// #
+/// # impl Model for AppData {}
+/// #
+/// # enum AppEvent {
+/// #     ToggleValue,
+/// # }
+/// #
+/// # let cx = &mut Context::new();
+/// #
+/// # AppData { value: false }.build(cx);
+/// # use vizia_core::fonts::unicode_names::CANCEL;
+///
+/// Switch::new(cx, AppData::value)
+///     .on_toggle(|cx| cx.emit(AppEvent::ToggleValue))
+///     .text(AppData::value.map(|flag| if *flag {CANCEL} else {""}));
+/// ```
+pub struct Switch {
+    on_toggle: Option<Box<dyn Fn(&mut EventContext)>>,
+}
+
+impl Switch {
+    /// Creates a new Switch.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use vizia_core::prelude::*;
+    /// #
+    /// # #[derive(Lens)]
+    /// # struct AppData {
+    /// #     value: bool,
+    /// # }
+    /// #
+    /// # impl Model for AppData {}
+    /// #
+    /// # let cx = &mut Context::new();
+    /// #
+    /// # AppData { value: false }.build(cx);
+    /// #
+    /// Switch::new(cx, AppData::value);
+    /// ```
+    pub fn new(cx: &mut Context, checked: impl Lens<Target = bool>) -> Handle<Self> {
+        //let checked = checked.get_val_fallible(cx).unwrap_or(false);
+        Self { on_toggle: None }
+            .build(cx, |cx| {
+                HStack::new(cx, |cx| {
+                    HStack::new(cx, |_|{}).class("switch-handle").hoverable(false);
```

Comment:
You can use an `Element` here instead of a `HStack` because it doesn't need to be a container.